server/quota/quota.go

```diff
@@ -102,14 +106,88 @@ func (m *Manager) check(ctx context.Context, namespace string, size int) error {
 		return ErrThroughputExceeded
 	}
 
-	return m.checkStorageSize(ctx, namespace, s, size)
+	return m.checkStorage(ctx, namespace, s, size)
+}
+
+func getDbSize(ctx context.Context, tenant *metadata.Tenant, tx transaction.Tx, dbName string) int64 {
+	db, err := tenant.GetDatabase(ctx, tx, dbName)
+	if err != nil {
+		ulog.E(err)
+	}
+	dbSize, err := tenant.DatabaseSize(ctx, db)
+	if err != nil {
+		ulog.E(err)
+	}
+	return dbSize
+}
+
+func getCollSize(ctx context.Context, tenant *metadata.Tenant, db *metadata.Database, coll *schema.DefaultCollection) int64 {
+	collSize, err := tenant.CollectionSize(ctx, db, coll)
+	if err != nil {
+		ulog.E(err)
+	}
+	return collSize
+}
+
+func (m *Manager) updateTenantSize(ctx context.Context, namespace string) {
+	if m.txMgr == nil {
+		return
+	}
+	tenant, err := m.tenantMgr.GetTenant(ctx, namespace, m.txMgr)
+	if err != nil {
+		ulog.E(err)
+		// Could not determine tenant, just exit
+		return
+	}
+
+	tx, err := m.txMgr.StartTx(ctx)
+	if err != nil {
+		ulog.E(err)
+		return
+	}
+
+	for _, dbName := range tenant.ListDatabases(ctx, tx) {
+		metrics.UpdateDbSizeMetrics(namespace, dbName, getDbSize(ctx, tenant, tx, dbName))
+		db, err := tenant.GetDatabase(ctx, tx, dbName)
```

Comment:
If you accept the database object in the `getDbSize` function, then you can make the `tenant.GetDatabase` call just once instead of twice.